src/nunit.analyzers.tests/MissingProperty/MissingPropertyCodeFixTests.cs

```diff
@@ -0,0 +1,94 @@
+        {
+            var code = TestUtility.WrapInTestMethod($@"
+                var collection = (IReadOnlyCollection<int>)new[] {{ 42 }};
+                Assert.That(collection, ↓Has.Length.EqualTo(1));",
+                additionalUsings: "using System.Collections.Generic;");
+
+            var fixedCode = TestUtility.WrapInTestMethod($@"
+                var collection = (IReadOnlyCollection<int>)new[] {{ 42 }};
+                Assert.That(collection, Has.Count.EqualTo(1));",
+                additionalUsings: "using System.Collections.Generic;");
+
+            AnalyzerAssert.CodeFix(analyzer, fix, expectedDiagnostic, code, fixedCode,
+                fixTitle: string.Format(CodeFixConstants.MissingPropertyDescriptionFormat, "Count"));
+        }
+
+        [Test]
+        public void VerifyCountToLengthCodeFix()
+        {
+            var code = TestUtility.WrapInTestMethod($@"
+                var collection = new[] {{ 42 }};
+                Assert.That(collection, ↓Has.Count.EqualTo(1));",
+                additionalUsings: "using System.Collections.Generic;");
+
+            var fixedCode = TestUtility.WrapInTestMethod($@"
+                var collection = new[] {{ 42 }};
+                Assert.That(collection, Has.Length.EqualTo(1));",
+                additionalUsings: "using System.Collections.Generic;");
+
+            AnalyzerAssert.CodeFix(analyzer, fix, expectedDiagnostic, code, fixedCode,
+                fixTitle: string.Format(CodeFixConstants.MissingPropertyDescriptionFormat, "Length"));
+        }
+
+        [Test]
+        public void NoCodeFixForLengthWhenCountPropertyDoesNotExist()
+        {
+            var testCode = TestUtility.WrapInTestMethod(@"
+                var actual = new [] {1, 2, 3}.Where(i => i > 1);
+                Assert.That(actual, ↓Has.Length.EqualTo(2));",
+                additionalUsings: "using System.Linq;");
+
+            AnalyzerAssert.NoFix(analyzer, fix, expectedDiagnostic, testCode);
+        }
+
+        [Test]
+        public void NoCodeFixForCountWhenLengthPropertyDoesNotExist()
+        {
+            var testCode = TestUtility.WrapInTestMethod(@"
+                var actual = new [] {1, 2, 3}.Where(i => i > 1);
+                Assert.That(actual, ↓Has.Count.EqualTo(2));",
+                additionalUsings: "using System.Linq;");
+
+            AnalyzerAssert.NoFix(analyzer, fix, expectedDiagnostic, testCode);
+        }
+
+        [Test]
+        public void NoCodeFixForHasMessage()
+        {
+            var testCode = TestUtility.WrapInTestMethod(@"
+                var actual = ""Can we consider string as message?..."";
+                Assert.That(actual, ↓Has.Message.EqualTo(2);");
+
+            AnalyzerAssert.NoFix(analyzer, fix, expectedDiagnostic, testCode);
+        }
+
+        [Test]
+        public void NoCodeFixForRandomMissingProperty()
+        {
+            var testCode = TestUtility.WrapInTestMethod(@"
+                var actual = 12345;
```

Comment:
suggestion: could you make the actual here to have `Count` property? To check that code fix is not available when property is present, but original constraint is not suitable?